pkg/psmdb/mongos.go

```diff
@@ -0,0 +1,361 @@
+package psmdb
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	api "github.com/percona/percona-server-mongodb-operator/pkg/apis/psmdb/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func MongosDeployment(cr *api.PerconaServerMongoDB) *appsv1.Deployment {
+	return &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name + "-" + "mongos",
+			Namespace: cr.Namespace,
+		},
+	}
+}
+
+func MongosDeploymentSpec(cr *api.PerconaServerMongoDB, operatorPod corev1.Pod) (appsv1.DeploymentSpec, error) {
+	ls := map[string]string{
+		"app.kubernetes.io/component": "mongos",
+		"app.kubernetes.io/instance":  cr.Name,
```

Comment:
can we add labels identical for mongod?
```suggestion
		"app.kubernetes.io/name": "percona-server-mongodb",
		"app.kubernetes.io/instance":  cr.Name,
		"app.kubernetes.io/component": "mongos",
		"app.kubernetes.io/managed-by": "percona-server-mongodb-operator"
		"app.kubernetes.io/part-of":    "percona-server-mongodb"
```